.github/workflows/build.yaml

```diff
@@ -26,162 +26,169 @@ jobs:
-           echo "PUBLISH_ARGS=--plugin sign-images" >> $GITHUB_ENV
-           docker trust key load $USER_KEY
-    - name: Set Push options
-      if: github.ref == 'refs/heads/master' && github.event_name == 'push'
-      run: echo "BUILD_ARGS=--push --no-spinner --live-output --only-target-package --pull" >> $GITHUB_ENV
+      - name: Login to DockerHub Registry
+        if: github.ref == 'refs/heads/master' && github.event_name == 'push'
+        run: echo ${{ secrets.DOCKER_PASSWORD }} | docker login -u ${{ secrets.DOCKER_USERNAME }} --password-stdin
+      - name: Load Docker Content Trust
+        if: github.ref == 'refs/heads/master' && github.event_name == 'push'
+        run: |
+          sudo cp -rf .github/sign-images /usr/bin/sign-images
+          sudo chmod +x /usr/bin/sign-images
+          PATH_KEYS=$HOME/.docker/trust/private
+          USER_KEY=$PATH_KEYS/7b1922fb6fa9046a2c64c16e1e557c092699854c23dedf653fab6faac00390a5.key
+          mkdir -p $PATH_KEYS
+          echo "${{ secrets.DOCKER_CONTENT_TRUST_PRIVKEY }}" > $USER_KEY
+          chmod 600 $USER_KEY
+          export DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE=${{ secrets.DOCKER_CONTENT_TRUST_USER_PASSPHRASE }}
+          echo "PATH_KEYS=$PATH_KEYS" >> $GITHUB_ENV
+          echo "DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE=$DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE" >> $GITHUB_ENV
+          echo "PUBLISH_ARGS=--plugin sign-images" >> $GITHUB_ENV
+          docker trust key load $USER_KEY
+      - name: Set Push options
+        if: github.ref == 'refs/heads/master' && github.event_name == 'push'
+        run: echo "BUILD_ARGS=--push --no-spinner --live-output --only-target-package --pull" >> $GITHUB_ENV
 
-    - name: Install deps
-      run: |
-        sudo -E make deps
+      - name: Install deps
+        run: |
+          sudo -E make deps
+          wget -qO- https://github.com/Itxaka/luet-mtree/releases/download/v0.0.2/luet-mtree-v0.0.2-Linux-x86_64.tar.gz | tar xvz --
+          sudo cp luet-mtree /usr/bin/luet-mtree
 
-    - name: Validate 🌳
-      run: |
-        make validate
+      - name: Validate 🌳
+        run: |
+          make validate
 
-    - name: Build packages 🔧
-      run: |
-        sudo -E make build
-        ls -liah $PWD/build
+      - name: Build packages 🔧
+        run: |
+          sudo -E make build
+          ls -liah $PWD/build
 
-    - name: Create repo
-      run: |
-        sudo -E make create-repo
-    - name: Upload results
-      uses: actions/upload-artifact@v2
-      with:
-        name: build-${{ matrix.flavor }}
-        path: build
-        if-no-files-found: error
-    - name: Upload signing logs
-      uses: actions/upload-artifact@v2
-      with:
-        name: sign-${{ matrix.flavor }}
-        path: /tmp/sign_images.log
-        if-no-files-found: warn
+      - name: Create repo
+        run: |
+          sudo -E make create-repo
+      - name: Generate manifests
+        run: |
+          for f in build/*tar*; do
```

Comment:
Can't we make this part of the same extension? I guess it should be trivial to make it also work as an internal plugin. In this way if someone else has to embed this in other kinda of CIs it's enough to pass by something like `--plugin luet-mtree` during publish/create

Reply:
I havent checked how luet works with plugins, but I'll have a look!

> Can't we make this part of the same extension? 

You mean, iterate over a dir/glob and generate manifests for all the files found?
Should be possible, but then we either autogenerate the output file based on the input name, or lose the possibility of the stdout output, unless I rework it to be under a flag.

I think it would be a good addition, yes. But I think I rather concentrate on the manifest cehck and rework this at a later stage :D